src/runtime/platform/LibraryLoader.cs

```diff
@@ -0,0 +1,153 @@
+using System;
+using System.Runtime.InteropServices;
+
+namespace Python.Runtime.Platform
+{
+    interface ILibraryLoader
+    {
+        IntPtr Load(string dllToLoad);
+
+        IntPtr GetFunction(IntPtr hModule, string procedureName);
+
+        bool Free(IntPtr hModule);
+    }
+
+    static class LibraryLoader
+    {
+        public static ILibraryLoader Get(OperatingSystemType os)
+        {
+            switch (os)
+            {
+                case OperatingSystemType.Windows:
+                    return new WindowsLoader();
+                case OperatingSystemType.Darwin:
+                    return new DarwinLoader();
+                case OperatingSystemType.Linux:
+                    return new LinuxLoader();
+                default:
+                    throw new Exception($"This operating system ({os}) is not supported");
+            }
+        }
+    }
+
+    class LinuxLoader : ILibraryLoader
+    {
+        private static int RTLD_NOW = 0x2;
```

Comment:
const?

Reply:
I think `static` is fine, `const` not being available at runtime has created problems in the past and makes debugging more annoying.